ir/src/constraints/list_comprehension.rs

```diff
@@ -0,0 +1,328 @@
+                            validate_access(i, *size)?;
+                            return Ok(Expression::VectorAccess(VectorAccess::new(
+                                ident.clone(),
+                                range.start() + i,
+                            )));
+                        }
+                        _ => Err(SemanticError::InvalidListComprehension(
+                            "Invalid type for a vector".to_string(),
+                        ))?,
+                    }
+                }
+            }
+        }
+    }
+    Ok(expression.clone())
+}
+
+/// Parses a named trace access in a list comprehension expression.
+///
+/// # Errors
+/// - Returns an error if the iterable is an identifier and that identifier does not correspond to
+///   a trace column.
+/// - Returns an error if the iterable is a range.
+/// - Returns an error if the iterable is a slice and that identifier does not correspond to a
+///   trace column.
+fn parse_named_trace_access(
+    named_trace_access: &NamedTraceAccess,
+    expression: &Expression,
+    lc: &ListComprehension,
+    symbol_table: &SymbolTable,
+    i: usize,
+) -> Result<Expression, SemanticError> {
+    for (member, iterable) in lc.context() {
+        if named_trace_access.name() == member.name() {
+            match iterable {
+                Iterable::Identifier(ident) => {
+                    let ident_type = symbol_table.get_type(ident.name())?;
+                    match ident_type {
+                        IdentifierType::TraceColumns(size) => {
+                            validate_access(i, size.size())?;
+                            return Ok(Expression::NamedTraceAccess(NamedTraceAccess::new(
+                                ident.clone(),
+                                i,
+                                NEXT_ROW,
+                            )));
+                        }
+                        _ => Err(SemanticError::InvalidListComprehension(
+                            "Iterable should be a trace column".to_string(),
+                        ))?,
+                    }
+                }
+                Iterable::Range(_) => {
+                    return Err(SemanticError::InvalidListComprehension(
+                        "Iterable cannot be of range type here".to_string(),
+                    ));
+                }
+                Iterable::Slice(ident, range) => {
+                    let ident_type = symbol_table.get_type(ident.name())?;
+                    match ident_type {
+                        IdentifierType::TraceColumns(trace_columns) => {
+                            validate_access(i, trace_columns.size())?;
+                            return Ok(Expression::NamedTraceAccess(NamedTraceAccess::new(
+                                ident.clone(),
+                                range.start() + i,
+                                NEXT_ROW,
+                            )));
```

Comment:
Why do we hard-code `NEXT_ROW` here? Shouldn't the offset come from `trace_columns`?

Reply:
You are right, it should come from `named_trace_access` here. Thanks.